io.openems.edge.evcs.spelsberg.smart/src/io/openems/edge/evcs/spelsberg/smart/EvcsSpelsbergSmartImpl.java

```diff
@@ -0,0 +1,343 @@
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_ENERGY_SESSION, new UnsignedWordElement(1502))),
+
+				new FC3ReadRegistersTask(1504, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_START_TIME, new UnsignedDoublewordElement(1504))),
+
+				new FC3ReadRegistersTask(1508, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_DURATION_SESSION, new UnsignedDoublewordElement(1508))),
+
+				new FC3ReadRegistersTask(1512, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_STOP_TIME, new UnsignedDoublewordElement(1512))),
+
+				new FC3ReadRegistersTask(2000, Priority.LOW,
+						m(EvcsSpelsbergSmart.ChannelId.CHARGE_SAVE_CURRENT_LIMIT, new UnsignedWordElement(2000))),
+
+				new FC3ReadRegistersTask(6000, Priority.HIGH,
+						m(EvcsSpelsbergSmart.ChannelId.LIFE_BIT, new UnsignedWordElement(6000))),
+
+				new FC6WriteRegisterTask(6000, m(EvcsSpelsbergSmart.ChannelId.LIFE_BIT, new UnsignedWordElement(6000))),
+
+				new FC16WriteRegistersTask(5000,
+						m(EvcsSpelsbergSmart.ChannelId.APPLY_CHARGE_POWER_LIMIT, new UnsignedDoublewordElement(5000))),
+
+				new FC6WriteRegisterTask(5004,
+						m(EvcsSpelsbergSmart.ChannelId.APPLY_CHARGE_CURRENT_LIMIT, new UnsignedWordElement(5004))));
+
+		addStatusCallback();
+		addPhaseDetectionCallback();
+		addPowerConsumptionCallback();
+		return modbusProtocol;
+	}
+
+	@Override
+	public String debugLog() {
+		return "Status: " + getStatus().getName() + " | " + "Charge Power Limit: "
+				+ getSetChargePowerLimit().orElse(null) + " | " + "Actual Charge Power: " + getChargePowerTotal()
+				+ " | " + "Charged Energy: " + getEnergySessionChannel().value();
+	}
+
+	@Override
+	public void logDebug(String message) {
+		if (this.config.debugMode()) {
+			this.logInfo(this.log, message);
+		}
+	}
+
+	@Override
+	public EvcsPower getEvcsPower() {
+		return this.evcsPower;
+	}
+
+	@Override
+	public int getConfiguredMinimumHardwarePower() {
+		return currentToPower(this.config.minHwCurrent());
+	}
+
+	@Override
+	public int getConfiguredMaximumHardwarePower() {
+		return currentToPower(this.config.maxHwCurrent());
+	}
+
+	@Override
+	public boolean getConfiguredDebugMode() {
+		return this.config.debugMode();
+	}
+
+	@Override
+	public boolean applyChargePowerLimit(int power) throws Exception {
+		setApplyChargePowerLimit(power);
+		return true;
+	}
+
+	@Override
+	public boolean pauseChargeProcess() throws Exception {
+		this.applyChargePowerLimit(0);
+		return true;
+	}
+
+	@Override
+	public boolean applyDisplayText(String text) throws OpenemsException {
+		return false;
+	}
+
+	@Override
+	public int getMinimumTimeTillChargingLimitTaken() {
+		return 30;
+	}
+
+	@Override
+	public ChargeStateHandler getChargeStateHandler() {
+		return this.chargeStateHandler;
+	}
+
+	@Override
+	public void handleEvent(Event event) {
+		if (!this.isEnabled()) {
+			return;
+		}
+		switch (event.getTopic()) {
+		case EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE:
+			writeHandler.run();
```

Comment:
I think this would be marked by our checkstyle file. (Add this. )
Explained here:
https://openems.github.io/openems.io/openems/latest/contribute/coding-guidelines.html#_tool_support

Reply:
I have added the checkstyle to the IDE and resolved the warnings showing up